### PR TITLE
feat: moved docker init out of E2E test file

### PR DESCRIPTION
**Pull request by commoddity**

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ jobs:
 
   test:
     name: Test
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-18.04
```
Comment (adshmh) [inline .github/workflows/build-test.yml:32]: Thank you for fixing the platform to a specific version!

File: Makefile [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
 
+test:
+	docker-compose -f ./tests/docker-compose.yml up -d --remove-orphans --build && \
+	go test ./... -count=1 -v && \
+	docker-compose -f ./tests/docker-compose.yml down --remove-orphans --rmi all -v
+
 test_e2e:
-	go test ./... -run E2E
+	docker-compose -f ./tests/docker-compose.yml up -d --remove-orphans --build && \
+	go test ./... -run E2E -count=1 -v && \
```
Comment (adshmh) [inline Makefile:11]: go test will return non-zero output if the test fails, blocking the next command, i.e. docker-compose down. Is that the intention here?

File: tests/e2e_test.go [hunk 3]
```diff
@@ -42,16 +41,6 @@ type PHDTestSuite struct {
 }
 
 func (t *PHDTestSuite) SetupSuite() {
-	_, err := exec.Command("docker", "compose", "up", "-d", "--build", "--force-recreate").Output()
```
Comment (adshmh) [inline tests/e2e_test.go:45]: great, thank you for removing the container setup from the test code!

**Review by adshmh — CHANGES_REQUESTED** [review]
Thank you for the PR! Looks good, just a few minor suggestions in the comments.
Also, please rebase to make sure only meaningful commits show up in the history.

File: Makefile (hunk 2 above)
Comment (commoddity) [reply Makefile:11]: Not intended, this has actually been a pain point for me when running the tests locally (ie. I have to run docker-compose down manually if the tests fail) so I'm going to look into how to get it to run regardless of `go test`s exit code.

Comment (commoddity) [reply Makefile]: Fixed now.

**Review by adshmh — APPROVED** [review]
Thank you for the PR. LGTM.
Two minor comments for future PRs:
- Merge conflicts should be amended to the correct commit, rather than being separate commits.
- Any formatting/linting needs its own commit (at-least when the modified file has no changes related to the PR).

**Pull request merged**